pkg/compact/compact.go

```diff
@@ -977,8 +973,10 @@ func (c *BucketCompactor) Compact(ctx context.Context) (rerr error) {
 							continue
 						}
 					}
-					// if block has out of order chunk, mark the block for no compaction and continue.
-					if IsOutOfOrderChunkError(err) {
+					// If block has out of order chunk and it has been configured to skip it,
+					// then we can mark the block for no compaction so that the next compaction run
+					// will skip it.
+					if IsOutOfOrderChunkError(err) && c.skipBlocksWithOutOfOrderChunks {
```

Comment:
I believe this is the right place where we should check for `skipBlocksWithOutOfOrderChunks`. If enabled (which means we skip it), then we do mark the block for no-compaction and we continue processing the next job from the queue. If disabled (which means we don't skip the block) then we should error out (done below).